superset-frontend/packages/superset-ui-chart-controls/src/types.ts

```diff
@@ -84,6 +84,12 @@ export interface Dataset {
   filter_select?: boolean;
   filter_select_enabled?: boolean;
   column_names?: string[];
+  catalog?: string;
+  schema?: string;
+  table_name?: string;
+  database?: Record<string, unknown>;
```

Comment:
Could we improve the typing here? We probably have a type for database defined somewhere

Reply:
@fisjac +1 for this